go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -197,6 +196,12 @@ func (fe *FileEntry) open(cnf *Mycnf, readOnly bool) (*os.File, error) {
 		if fd, err = os.Open(name); err != nil {
 			return nil, vterrors.Wrapf(err, "cannot open source file %v", name)
 		}
+		// hint to the kernel that the intent is to read this file sequentially
+		if err := fadviseSequential(fd); err != nil {
+			// close the open fd since we're not using it anymore
+			fd.Close()
+			return nil, vterrors.Wrapf(err, "could not fadvise sequential read for %v", name)
+		}
```

Comment:
One alternative here, and not sure if there's a preference, this whole block can swapped out to a `openForSequential` and have the two implementation that both encapsulate the os.Open call as well.

```go
//go:build linux
func openForSequential(name string) (*os.File, error) {
  f, err := os.Open(name)
  if err != nil {
    return nil, err
  }
  fstat, err := f.Stat()
  if err != nil {
      f.Close()
      return nil, err
  }
  if err := unix.Fadvise(int(f.Fd()), 0, fstat.Size(), unix.FADV_SEQUENTIAL); err != nil {
    f.Close()
    return nil, err
  }
  return f, nil
}
```

```go
//go:build !linux
func openForSequential(name string) (*os.File, error) {
  return os.Open(name)
}
```

I dunno if the project would have a preference for this instead.

Reply:
I like this approach.

Reply:
@mattrobenolt Seems like a good approach for this to have those separate implementations then.

Reply:
Technically the build tag is likely not `linux` though (see for example https://man.freebsd.org/cgi/man.cgi?query=posix_fadvise&sektion=2&manpath=FreeBSD+8.3-RELEASE)? But we only really support Vitess properly here and it's purely an optimization so Linux specific also seems fine if it's problematic to have a better build tag.

Reply:
From what I can tell, the `golang.org/x/sys/unix` only exposes `unix.Fadvise` for linux targets.

Also a GitHub search for this, every usage I can find also explicitly only targets linux: https://github.com/search?q=unix.Fadvise+language%3AGo&type=code&l=Go

I tried to look for a more generic build tag, but I don't think there's anything applicable, unless we directly used `syscall.Syscall6` or whatever directly, and I don't think that's worth it. Seems more like if there's freebsd support, that should get into the `unix` package.

Reply:
Will swap this out tho for a generic `openForSequential`.